src/query/service/src/interpreters/access/privilege_access.rs

```diff
@@ -544,6 +567,10 @@ impl AccessChecker for PrivilegeAccess {
                     .await?;
             }
             Plan::UseDatabase(plan) => {
+                let session = self.ctx.get_current_session();
+                if self.has_ownership(&session, &GrantObject::Database(catalog_name.clone(), plan.database.clone())).await? {
```

Comment:
here we use self.has_ownership, it means do both read and write ownership?

Reply:
https://github.com/datafuselabs/databend/blob/e3de5caaef711129540ae0c2bbc5776b2bb01c27/src/query/service/src/sessions/session_privilege_mgr.rs#L275

It will call this.

Reply:
yep, an role whom have ownership on an object means she can do anything like read/write/drop to this object.